shop/shop/src/main/java/ro/msg/learning/shop/exception/StrategyNotApplicableException.java

```diff
@@ -0,0 +1,11 @@
+package ro.msg.learning.shop.exception;
+
+public class StrategyNotApplicableException extends RuntimeException{
+    private String message;
```

Comment:
I don't think you need a local message attribute and neither an empty constructor